ospsuite-r-specifics/r-code-structure.md

```diff
@@ -0,0 +1,250 @@
+# OSPSuite-R Package Structure
+
+## Introduction
+
+In this part of the documentation we will talk about the specifics of the [OSPSuite-R](https://github.com/Open-Systems-Pharmacology/OSPSuite-R) package. This is a package that offers the functionalities of OSPSuite to the R language. We will analyze its elements and code structure, as well as the components that enable us to interface between the OSPSuite codebase in the .NET universe and the R programming language. 
+
+## OSPSuite-R communication with .NET
+
+As mentioned, through the OSPSuite-R package we get OSPSuite functionalities available in the R programming language. Those OSPSuite functionalities though have been developed and exist in the .NET universe. In order to provide them in R we have to enable communication between R and .NET. To do this we are currently using the [rClr package](https://github.com/Open-Systems-Pharmacology/rClr). This package allows the communication between R and .NET using C++ as an intermediate layer. .NET can communicate with C++ using a custom native host and C++ can then communicate with R through the R .C interface. This is why we need to have the rClr package installed in R for OSPSuite-R to work. Using rClr we can load the dlls produced from the .NET code and use them.
```

Comment:
Add here a high-level overview of .NET libraries that are used. That the API between `OSPSuite.Core` is implemented in `OSPSuite.R`, additionally, `PKSim.R` for calling PK-Sim specific functionalities.

Reply:
@PavelBal .NET libraries here you mean the OSPSuite ones, or the other .NET ones (npoi and so on)?